packages/react-app/src/constants.js

```diff
@@ -19,10 +19,16 @@ Note: this is not used when you are in the local hardhat chain.
 */
 export const RPC_POLL_TIME = 30000;
 
-const localRpcUrl = process.env.REACT_APP_CODESPACES
+let localRpcUrl = process.env.REACT_APP_CODESPACES
   ? `https://${window.location.hostname.replace("3000", "8545")}`
   : "http://" + (global.window ? window.location.hostname : "localhost") + ":8545";
 
+// When running inside docker, we need to use the HARDHAT_HOST env variable
+// in order to access the container running hardhat
+if (process.env.HARDHAT_HOST) {
+  localRpcUrl = `http://${process.env.HARDHAT_HOST}:8545`;
+}
+
```

Comment:
This can be skipped I think because the container ports are exposed on the same network

Reply:
wow, I think I may not need HARDHAT_HOST at all? I thought because app and chain were running in different containers I'd need to set the host for them to talk to each other. But if localhost works then HARDHAT_HOST isn't need at all, right?

Thanks for the review, btw!